File: mac/Classes/LRActionVersion.m

#import "LRActionVersion.h"
@import PackageManagerKit;


@interface LRActionVersion ()

@end


@implementation LRActionVersion

- (instancetype)initWithType:(Action *)type manifest:(LRActionManifest *)manifest packageSet:(LRPackageSet *)packageSet {
    if (self = [super init]) {
        _type = type;
        _manifest = manifest;
        _packageSet = packageSet;
    }
    return self;
}

- (LRVersion *)primaryVersion {
    LRPackage *package = [_packageSet.packages firstObject];
    return package.version;
}

@end

File: mac/LRActionKit/LRActionKit/Action/LRActionVersion.swift
import Foundation
import PackageManagerKit
import PiiVersionKit
import LRCommons

public class LRActionVersion: NSObject {

    public var action: Action
    public var manifest: LRActionManifest
    public var packageSet: LRPackageSet

    public init(action: Action, manifest: LRActionManifest, packageSet: LRPackageSet) {
        self.action = action
        self.manifest = manifest
        self.packageSet = packageSet
        super.init()

        if packageSet.packages.isEmpty {
            fatalError("LRActionVersion.init with an empty packageSet")
        }
    }

    public var primaryVersion: LRVersion {
        let packages = packageSet.packages as! [LRPackage]
        return packages[0].version
    }

    public override var description: String {
        return primaryVersion.description
    }

}
